### PR TITLE
add five design pattern

### PR DESCRIPTION
**Pull request by shantontsun**
just demo, there will be many bugs and improvements
1.proxy
2.decorator
3.null
4.adapter
5.bridge

### CODE REVIEW

File: src/main/java/org/derekdaisy/happyfarm/BedRoom.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import org.derekdaisy.happyfarm.field.Room;
+
+/**
+ * @author XuantangCun E-mail:Tang_zhu0@163.com
+ * @version 创建时间：2017-10-24
+ *
+ */
+public class BedRoom extends Room {
+
+    public BedRoom(){
```
Comment (WinDerek) [inline src/main/java/org/derekdaisy/happyfarm/BedRoom.java:12]: Bad code style:weary:

File: src/main/java/org/derekdaisy/happyfarm/BedRoom.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        goods.add("bed");
+        goods.add("mirror");
+    }
+
+    @Override
+    public void show() {
+        System.out.println("this room have good here: ");
+        for(String str : goods){
```
Comment (WinDerek) [inline src/main/java/org/derekdaisy/happyfarm/BedRoom.java:20]: Bad code style:weary:

File: src/main/java/org/derekdaisy/happyfarm/CabinetRoomDecorator.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import org.derekdaisy.happyfarm.field.Room;
+/**
+ * @author XuantangCun E-mail:Tang_zhu0@163.com
+ * @version 创建时间：2017-10-24
+ *
+ */
+public class CabinetRoomDecorator extends Room {
+
+    private Room room;
+    public CabinetRoomDecorator(Room room){
```
Comment (WinDerek) [inline src/main/java/org/derekdaisy/happyfarm/CabinetRoomDecorator.java:12]: Bad code style:weary:

File: src/main/java/org/derekdaisy/happyfarm/AndroidSimulator.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package org.derekdaisy.happyfarm;
+
+/**
+ * @author XuantangCun E-mail:Tang_zhu0@163.com
+ * @version 创建时间：2017-10-24
```
Comment (WinDerek) [inline src/main/java/org/derekdaisy/happyfarm/AndroidSimulator.java:5]: Bad code style:weary:

File: src/main/java/org/derekdaisy/happyfarm/BedRoom.java (hunk 1 above)
Comment (shantontsun) [reply src/main/java/org/derekdaisy/happyfarm/BedRoom.java:12]: ｆｕｎｃｋ

**Comment by WinDerek** [comment]
Please adhere to the code style specification shared in the QQ group. @wzes

**Pull request closed without merging**